src/resolvers/Mutation/createPost.ts

```diff
@@ -108,5 +109,10 @@ export const createPost: MutationResolvers["createPost"] = async (
   }
 
   // Returns createdPost.
-  return createdPost.toObject();
+  return {
+    ...createdPost.toObject(),
+    imageUrl: createdPost.imageUrl
+      ? `${BASE_URL}${uploadImageFileName}`
+      : undefined,
```

Comment:
Make the value after `:` null. `undefined` is kinda a wack thing in javascript which should be avoided whenever we can. For nullable values `null` should be explicitly used. On client side, `imageUrl` can either be a `string` or `null` if the client asked for it in their graphql query explicitly. Graphql would never return `undefined`.